### PR TITLE
fix: duplicate search content when `/README` or `/` exists in the sidebar

### PR DESCRIPTION
**Pull request by sy-records**
**Summary**

fix: duplicate search content when `/README` or `/` exists in the sidebar.

image

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Repo settings
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

### CODE REVIEW

File: src/plugins/search/search.js [hunk 1]
```diff
@@ -236,8 +236,13 @@ export function init(config, vm) {
...
-  } else {
+    if (
+      paths.indexOf(namespaceSuffix + '/') === -1 &&
+      paths.indexOf(namespaceSuffix + '/README') === -1
+    ) {
+      paths.unshift(namespaceSuffix + '/');
+    }
+  } else if (paths.indexOf('/') === -1 && paths.indexOf('/README') === -1) {
     paths.unshift('/');
   }
```
Comment (jhildenbiddle) [inline src/plugins/search/search.js:247]: One suggestion is to convert these if/else conditions to `isXXX` variables so the code is easier to read.

For example, instead of this:

```js
if (
  paths.indexOf(namespaceSuffix + '/') === -1 &&
  paths.indexOf(namespaceSuffix + '/README') === -1
) {
 // ...
}
```

Consider:

```js
const isCondition1 = paths.indexOf(namespaceSuffix + '/') === -1;
const isCondition2 = paths.indexOf(namespaceSuffix + '/README') === -1;

if (isCondition1 && isCondition1) {
  // ...
)
```

**Review by jhildenbiddle — CHANGES_REQUESTED** [review]
I tested this PR locally and still encounter the issue. To reproduce the error:

1. Add the following line to `_sidebar.md`:
   ```
   - [README](README)
   ```
1. Launch test server with the following CLI command:
  ```bash
   node ./test/config/server.js --start
   ```
1. Search for "generator"

Results

Screen Shot 2020-11-20 at 2 34 47 PM

We should also add tests for this scenario (no duplicates).

**Comment by sy-records** [comment]
> I tested this PR locally and still encounter the issue. To reproduce the error:
> 
> 1. Add the following line to `_sidebar.md`:
>    ```
>    - [README](README)
>    ```
> 2. Launch test server with the following CLI command:
> 
> ```shell
>  node ./test/config/server.js --start
> ```
> 
> 1. Search for "generator"
> 
> Results
> 
> Screen Shot 2020-11-20 at 2 34 47 PM

 You need to run `npm run build:js` first, then delete the local storage cache.

image

File: src/plugins/search/search.js (hunk 1 above)
Comment (sy-records) [reply src/plugins/search/search.js:246]: done.

**Comment by jhildenbiddle** [comment]
@sy-records --

Fixed the test. Required updating the Playwright config to clear local storage before each test. I assumed that would have been done before, but apparently not. See the following branch: https://github.com/docsifyjs/docsify/tree/pullrequests/sy-records/fix-search

**Review by jhildenbiddle — APPROVED** [review]
LGTM!

**Review by anikethsaha — APPROVED** [review]
